### PR TITLE
Add --with-socket-path=... and --with-cache-dir=... to configure

### PR DESCRIPTION
**Pull request by josuedhg**
also add an sleep to the testing_fuse test

Signed-off-by: Josue David Hernandez Gutierrez <josue.d.hernandez.gutierrez@intel.com>

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -38,6 +36,20 @@ else
...
+            [dir="/run/clr-debug-info"])
+test -z "${dir}" && dir=/run/clr-debug-info
+AC_DEFINE_UNQUOTED([SOCKET_PATH], ["${dir}"], [path to create unix socket @<:@default=/run/clr-debug-info@:>@])
+
+dir=""
+AC_ARG_WITH([cache-dir], AS_HELP_STRING([--with-cache-dir=CACHE_DIR],
+            [path to cache downloaded content @<:@default=/var/cache/debuginfo@:>@]), [dir="${withval}"],
+            [dir="/var/cache/debuginfo"])
+test -z "${dir}" && dir=/var/cache/debuginfo
+AC_DEFINE_UNQUOTED([CACHE_DIR], ["${dir}"], [path to cache downloaded content @<:@default=/var/cache/debuginfo@:>@])
```
Comment (phmccarty) [inline configure.ac]: Can you also update the `AC_MSG_RESULT` macro below to print the values for the two paths? You will want to use different shell variables to store the paths (instead of reusing `dir`) so that the values can be printed separately.

File: tests/testing_fuse.c [hunk 2]
```diff
@@ -147,6 +147,7 @@ int main(void)
                 char *args[] = {"./clr_debug_fuse", NULL};
                 execvp("./clr_debug_fuse", args);
         } else if (p > 0) {
+                sleep(1);
```
Comment (phmccarty) [inline tests/testing_fuse.c]: unrelated change?

File: src/server.c [hunk 3]
```diff
@@ -338,8 +339,9 @@ static int curl_get_file(const char *url, const char *prefix, time_t timestamp)
 
                 free(command); /* reuse */
                 if (asprintf(&command,
-                             "tar -C /var/cache/debuginfo/%s --no-same-owner "
+                             "tar -C %s/%s --no-same-owner "
                              "--no-same-permissions -xf %s",
+                             CACHE_DIR,
```
Comment (phmccarty) [inline src/server.c:344]: These changes looks fine, but the hardcoded paths in `main()` also need to be updated (line 477).

**Review by phmccarty — COMMENTED** [review]
I just have a few comments inline.

Also, the `debuginfo.conf` has hardcoded paths. I would recommend changing that file to be a template and using `sed` to generate the file instead. This is how telemetrics-client generates its conf files (https://github.com/clearlinux/telemetrics-client/blob/master/src/data/local.mk)

File: tests/testing_fuse.c (hunk 2 above)
Comment (josuedhg) [reply tests/testing_fuse.c]: yes, I can put it in other commit.

File: src/server.c (hunk 3 above)
Comment (josuedhg) [reply src/server.c:344]: sorry I didn't see them, changing

**Review by phmccarty — APPROVED** [review]
LGTM, thanks!
